File: fixtures/javascript/ledger.js
export function parseEntry(raw) {
  const [title = "", ...tags] = raw.split("#").map((part) => part.trim());
  if (!title) {
    throw new TypeError("title is required");
  }
  return Object.freeze({ title, tags: tags.filter(Boolean), settled: false });
}

export function summarize({ title, tags, settled }) {
  const mark = settled ? "✓" : "·";
  return `${mark} ${title} [${tags.join(", ")}]`;
}

File: fixtures/python/ledger.py
from dataclasses import dataclass, field


@dataclass(frozen=True, slots=True)
class Entry:
    title: str
    tags: tuple[str, ...] = field(default_factory=tuple)
    settled: bool = False

    def summary(self) -> str:
        mark = "✓" if self.settled else "·"
        return f"{mark} {self.title} [{', '.join(self.tags)}]"


def parse_entry(raw: str) -> Entry:
    title, *tags = (part.strip() for part in raw.split("#"))
    if not title:
        raise ValueError("title is required")
    return Entry(title=title, tags=tuple(filter(None, tags)))
